contracts/token/NonFungibleToken.sol

```diff
@@ -0,0 +1,289 @@
+    return tokenApprovals[_tokenId];
+  }
+
+  /**
+   * @dev Tells whether an operator is approved by a given owner
+   * @param _owner owner address which you want to query the approval of
+   * @param _operator operator address which you want to query the approval of
+   * @return bool whether the given operator is approved by the given owner
+   */
+  function isOperatorApprovedFor(address _owner, address _operator) public view returns (bool) {
+    return operatorApprovals[_owner][_operator];
+  }
+
+  /**
+  * @dev Transfers the ownership of a given token ID to another address
+  * @param _to address to receive the ownership of the given token ID
+  * @param _tokenId uint256 ID of the token to be transferred
+  */
+  function transfer(address _to, uint256 _tokenId) public onlyOwnerOf(_tokenId) {
+    clearApprovalAndTransfer(msg.sender, _to, _tokenId);
+  }
+
+  /**
+   * @dev Transfer the ownership of a given token ID to another contract address and calls it with given data
+   * @param _to address to receive the ownership of the given token ID
+   * @param _tokenId uint256 ID of the token to be transferred
+   * @param _data bytes of data to be sent when the receiving address gets called
+   */
+  function transfer(address _to, uint _tokenId, bytes _data) public onlyOwnerOf(_tokenId) {
+    clearApprovalAndTransfer(msg.sender, _to, _tokenId);
+    require(_to.call.value(0)(_data));
+  }
+
+  /**
+  * @dev Approves an address to claim ownership of any token owned by the msg.sender
+  * @param _to operator address to set their approval
+  * @param _approved representing the status of the approval
+  */
+  function approve(address _to, bool _approved) public {
+    require(_to != msg.sender); // we are reverting here following the same behaviour as approve does
+    operatorApprovals[msg.sender][_to] = _approved;
+  }
+
+  /**
+  * @dev Approves another address to claim for the ownership of the given token ID
+  * @param _to address to be approved for the given token ID
+  * @param _tokenId uint256 ID of the token to be approved
+  */
+  function approve(address _to, uint256 _tokenId) public onlyOwnerOf(_tokenId) {
+    address owner = ownerOf(_tokenId);
+    require(_to != owner);
+    if(approvedFor(_tokenId) == 0 && _to == 0) return;
+
+    tokenApprovals[_tokenId] = _to;
+    Approval(owner, _to, _tokenId);
+  }
+
+  /**
+  * @dev Claims the ownership of a given token ID
+  * @param _tokenId uint256 ID of the token being claimed by the msg.sender
+  */
+  function takeOwnership(uint256 _tokenId) public onlyApprovedFor(_tokenId) {
+    clearApprovalAndTransfer(ownerOf(_tokenId), msg.sender, _tokenId);
+  }
+
+  /**
+  * @dev Claims the ownership of a given token ID for a given recipient
+  * @param _to address which you want to transfer the token to
+  * @param _tokenId uint256 ID of the token being claimed by the msg.sender
+  */
+  function takeOwnershipFor(address _to, uint256 _tokenId) public onlyApprovedFor(_tokenId) {
```

Comment:
Hmmm, this name... But I see that this is the least agreed upon part in the ERC.

Did we make the name up or steal it from somewhere else?